src/main/scala/it/unibo/authsim/library/dsl/cryptography/encrypter/symmetric/CaesarCipher.scala

```diff
@@ -20,11 +20,11 @@ object CaesarCipherCipher:
   /**
    * Basic implementation of an encrypter which use Caesar Cipher algorithm for the cryptographic operation
    */
-  case class CaesarCipherImpl() extends Cipher:
+  case class CaesarCipherImpl() extends Cipher with SymmetricCipher:
```

Comment:
You should replace `CaesarCipherImpl` with `BasiCaesarCipher`  or make it `private`